ddmark/validation_integration_test.go

```diff
@@ -0,0 +1,184 @@
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+var _ = Describe("Validation Integration Tests", func() {
+	Context("Minimum/Maximum Markers", func() {
+		It("checks out valid values", func() {
+			var minmaxValidYaml string = `
+minmaxtest:
+  intfield: 6
+  pintfield: 6
+`
+			errorList := validateString(minmaxValidYaml)
+			Expect(errorList).To(HaveLen(0))
+		})
+		It("rejects empty pointer value", func() {
+			var minmaxValidYaml string = `
+minmaxtest:
+  intfield: 6
+  pintfield: 
+`
+			errorList := validateString(minmaxValidYaml)
+			Expect(errorList).To(HaveLen(0))
+		})
+		It("checks out valid values", func() {
+			var minmaxValidYaml string = `
+minmaxtest:
+  intfield: 5
+  pintfield: 10
+`
+			errorList := validateString(minmaxValidYaml)
+			Expect(errorList).To(HaveLen(0))
+		})
+		It("rejects invalid values", func() {
+			var minmaxInvalidYaml string = `
+minmaxtest:
+  intfield: 4
+  pintfield: 11
+`
+			errorList := validateString(minmaxInvalidYaml)
+			Expect(errorList).To(HaveLen(2))
+		})
+	})
+
+	Context("Required Marker", func() {
+		It("rejects on all but one missing fields", func() {
+			var requiredOneFieldYaml string = `
+requiredtest:
+  intfield: 1
+`
+			errorList := validateString(requiredOneFieldYaml)
+			Expect(errorList).To(HaveLen(5))
+		})
+		It("rejects on all but one missing fields", func() {
+			var requiredOneFieldYaml string = `
+requiredtest:
+  intfield: 1
+  pintfield: 
+`
+			errorList := validateString(requiredOneFieldYaml)
+			Expect(errorList).To(HaveLen(5))
+		})
+		It("rejects and counts all 4 missing fields", func() {
+			var requiredNoFieldYaml string = `
+requiredtest:
+  intfield: 1
+  pintfield: 0
+`
+			errorList := validateString(requiredNoFieldYaml)
+			Expect(errorList).To(HaveLen(4))
+		})
+		It("rejects and counts all 5 missing fields", func() {
+			var requiredNoField2Yaml string = `
+requiredtest:
+  pintfield: 1
+  pstructfield:
+    a:
+`
+			errorList := validateString(requiredNoField2Yaml)
+			Expect(errorList).To(HaveLen(4))
+		})
+		It("checks out on valid file", func() {
+			var requiredValidYaml string = `
+requiredtest:
+  intfield: 1
+  pintfield: 0
+  strfield: a
+  pstrfield: ""
+  structfield:
+    a: 1
+  pstructfield:
+    a: 1
+`
+			errorList := validateString(requiredValidYaml)
+			Expect(errorList).To(HaveLen(0))
+		})
+	})
+
+	Context("Enum Marker", func() {
+		It("checks out valid values", func() {
+			var enumCorrectYaml string = `
+enumtest:
+  strfield: aa
+  pstrfield: bb
+  intfield: 1
+  pintfield: 2
+`
+			errorList := validateString(enumCorrectYaml)
+			Expect(errorList).To(HaveLen(0))
+		})
+		It("rejects invalid values", func() {
+			var enumCorrectYaml string = `
+enumtest:
+  strfield: notinenum
+  pstrfield: notinenum
+  intfield: 4
+  pintfield: 4
+`
+			errorList := validateString(enumCorrectYaml)
+			Expect(errorList).To(HaveLen(4))
+		})
+	})
+
+	Context("ExclusiveFields Marker", func() {
```

Comment:
Ditto about `LinkedFields` tests here

Reply:
Will be included in the next PR (validation-to-ddmark branch)